### PR TITLE
Azfw ipgroups

### PR DESCRIPTION
**Pull request by ssripadham**
## Description

    - `Azure Firewall will support IpGroups can be used in the network/application/dnat rules as described below.

- As a source or destination address in AZFW network rules
- As a source address in AZFW application rules
- As a source address in DNAT rules. `

**1. Azure Firewall Application Rules**

         "AzureFirewallApplicationRule": {
             "properties": {
               "name": {
                 "type": "string",
                 "description": "Name of the application rule."
               },
               "description": {
                 "type": "string",
                 "description": "Description of the rule."
               },
               ...
               "sourceIpGroups": {
                 "type": "array",
                 "description": "List of source IP Groups for this rule.",
                 "items": {
                   "type": "string"
                 }
               },
           ...
         }
        }

**2. Azure Firewall  NAT Rules**

       "AzureFirewallNatRule": {
             "properties": {
               "name": {
                 "type": "string",
                 "description": "Name of the NAT rule."
               },
               "description": {
                 "type": "string",
                 "description": "Description of the rule."
               },
              ...

        "sourceIpGroups": {
          "type": "array",
          "description": "List of source IP Groups for this rule.",
          "items": {
            "type": "string"
          }
        }
           …
       }
       }

**3. Azure Firewall  Network Rules**

        "AzureFirewallNetworkRule": {
             "properties": {
               "name": {
                 "type": "string",
                 "description": "Name of the network rule."
               },
               "description": {
                 "type": "string",
                 "description": "Description of the rule."
               },
              ...
               "sourceIpGroups": {
                 "type": "array",
                 "description": "List of source IP Groups for this rule.",
                 "items": {
                   "type": "string"
                 }
               },
               "destinationIpGroups": {
                 "type": "array",
                 "description": "List of destination IP Groups for this rule.",
                 "items": {
                   "type": "string"
                 }
               },
           ...
        }
       }

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by VeryEarly** [comment]
Hi @ssripadham ,

Please fix the CI failures.

File: src/Network/Network/help/New-AzFirewallApplicationRule.md [hunk 1]
```diff
@@ -15,9 +15,10 @@ Creates a Firewall Application Rule.
 
 ### TargetFqdn (Default)
 ```
-New-AzFirewallApplicationRule -Name <String> [-Description <String>] [-SourceAddress <String[]>]
- -TargetFqdn <String[]> -Protocol <String[]> [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
- [<CommonParameters>]
+New-AzFirewallNetworkRule -Name <String> [-Description <String>] [-SourceAddress <String[]>]
+ [-SourceIpGroup <String[]>] [-DestinationAddress <String[]>] [-DestinationIpGroup <String[]>]
+ [-DestinationFqdn <String[]>] -DestinationPort <String[]> -Protocol <String[]>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
[the remark this answers is not in the document]
Comment (ssripadham) [reply src/Network/Network/help/New-AzFirewallApplicationRule.md]: Thanks. I have fixed this.

**Comment by VeryEarly** [comment]
Hi @ssripadham ,

Some test cases need to be updated after your changes.

**Comment by wyunchi-ms** [comment]
Closing this since there is a new pr https://github.com/Azure/azure-powershell/pull/10668.

**Pull request closed without merging**